do/apps.go

```diff
@@ -119,6 +121,86 @@ func (s *appsService) Delete(appID string) error {
 	return err
 }
 
+func (s *appsService) Detect(source string, sha string, name string, branch string, autoDeploy bool) (*godo.AppSpec, error) {
```

Comment:
I think we should consider changing this signature to only inline required parameters and put the rest in an options struct. e.g.:

```
type DetectOpts struct {
  Name string // random name generated by default
  Sha string // latest sha in branch by default
  AutoDeploy bool // False by default
}

func (s *appsService) Detect(source, branch string, opts DetectOpts) (*godo.AppSpec, error) {
```